commit 01684e72f16727e6ae0aeb1392f478e11ec5b8f7
Author: Arnd Bergmann <arnd@arndb.de>
Date:   Fri Feb 2 15:56:19 2018 +0100

    x86/error_inject: Make just_return_func() globally visible
    
    With link time optimizations enabled, I get a link failure:
    
      ./ccLbOEHX.ltrans19.ltrans.o: In function `override_function_with_return':
      <artificial>:(.text+0x7f3): undefined reference to `just_return_func'
    
    Marking the symbol .globl makes it work as expected.
    
    Signed-off-by: Arnd Bergmann <arnd@arndb.de>
    Acked-by: Masami Hiramatsu <mhiramat@kernel.org>
    Acked-by: Thomas Gleixner <tglx@linutronix.de>
    Cc: Alexei Starovoitov <ast@kernel.org>
    Cc: Josef Bacik <jbacik@fb.com>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Cc: Nicolas Pitre <nico@linaro.org>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Fixes: 540adea3809f ("error-injection: Separate error-injection from kprobe")
    Link: http://lkml.kernel.org/r/20180202145634.200291-3-arnd@arndb.de
    Signed-off-by: Ingo Molnar <mingo@kernel.org>

diff --git a/arch/x86/lib/error-inject.c b/arch/x86/lib/error-inject.c
index 7b881d03d0dd..3cdf06128d13 100644
--- a/arch/x86/lib/error-inject.c
+++ b/arch/x86/lib/error-inject.c
@@ -7,6 +7,7 @@ asmlinkage void just_return_func(void);
 
 asm(
 	".type just_return_func, @function\n"
+	".globl just_return_func\n"
 	"just_return_func:\n"
 	"	ret\n"
 	".size just_return_func, .-just_return_func\n"